crypto/rand/rand_lib.c

```diff
@@ -373,13 +378,18 @@ RAND_POOL *rand_pool_new(int entropy_requested, size_t min_len, size_t max_len)
     pool->max_len = (max_len > RAND_POOL_MAX_LENGTH) ?
         RAND_POOL_MAX_LENGTH : max_len;
 
-    pool->buffer = OPENSSL_secure_zalloc(pool->max_len);
+    if (secure)
+        pool->buffer = OPENSSL_secure_zalloc(pool->max_len);
+    else
+        pool->buffer = OPENSSL_malloc(pool->max_len);
```

Comment:
Why is the memory cleared on only one of the two branches?  I'd be tempted to clear on neither.

BTW: secure memory is allocated cleared if it exists, if not it falls back to the normal malloc.

Reply:
I hope the secure_zalloc does not clear it twice.

Reply:
It doesn't clear twice, secure_zalloc just calls secure_malloc.  Secure memory is always zero when not allocated (except for the pointers between free blocks which get cleared).